istioctl/cmd/proxystatus_v2_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+
+	"istio.io/istio/pilot/pkg/bootstrap"
+	"istio.io/istio/pilot/test/util"
+	"istio.io/istio/pkg/test/env"
+	pilot_util "istio.io/istio/tests/util"
+)
+
+func TestProxyStatusV2(t *testing.T) {
+	_, tearDown := pilot_util.EnsureTestServer(func(args *bootstrap.PilotArgs) {
+		// TODO remove any not needed or switch to something that will let istio.io/connections have data?
+		// TODO Verify this won't pull from my true cluster
+		args.Plugins = bootstrap.DefaultPlugins
+		args.Config.FileDir = env.IstioSrc + "/tests/testdata/networking/sidecar-ns-scope"
+		args.Mesh.MixerAddress = ""
+		args.MeshConfig = nil
+		args.Mesh.ConfigFile = env.IstioSrc + "/tests/testdata/networking/sidecar-ns-scope/mesh.yaml"
+		args.Service.Registries = []string{}
+	})
+	defer tearDown()
+
+	// TODO This fails with warn, ads	ADS: Unknown watched resources
+	// node:<id:"sidecar~1.2.3.4~test-1.default~default.svc.cluster.local" ...
+	// Why?
```

Comment:
@costinm @howardjohn Can you comment on why the test case times out?  The same code works with a running Istiod.

Reply:
How is the test initialized ? Does it use bootstrap or special code ? If it did use bootstrap - the 'events' feed should be known.

Reply:
@costinm The test was initialized by some code I cut-and-pasted without understanding from _lds_test.go_.  I even kept the _tests/testdata/networking/sidecar-ns-scope/configs.yaml_ directory to set up the test server.

Questions:
- Is `pilot_util.EnsureTestServer()` the correct way to set up a unit test for `istio.io/connections`?
- Is the test case supposed to supply yaml to initialize the data?  I didn't supply any pods, so I expected to get back zero `connections` and not a timeout.

Reply:
The PR has a test case - it inits and connect once, then makes a second connection.

From the logs on your tests - either the "generator" metadata was not set, or the istiod initialization didn't match. I breakpoint is the best way to handle this - the tests run great on
a VM in Goland.